L1Trigger/TrackFindingTracklet/src/CleanTrackMemory.cc

```diff
@@ -0,0 +1,114 @@
+#include "L1Trigger/TrackFindingTracklet/interface/CleanTrackMemory.h"
+#include "L1Trigger/TrackFindingTracklet/interface/Tracklet.h"
+#include "L1Trigger/TrackFindingTracklet/interface/slhcevent.h"
+
+using namespace std;
+using namespace Trklet;
+
+CleanTrackMemory::CleanTrackMemory(
+    string name, const Settings* const settings, unsigned int iSector, double phimin, double phimax)
+    : MemoryBase(name, settings, iSector) {
+  phimin_ = phimin;
+  phimax_ = phimax;
+}
+
+bool CleanTrackMemory::foundTrack(ofstream& outres, L1SimTrack simtrk) {
+  bool match = false;
+  double phioffset = phimin_ - (phimax_ - phimin_) / 6.0;
+  for (unsigned int i = 0; i < tracks_.size(); i++) {
+    match = match || tracks_[i]->foundTrack(simtrk, phimin_);
+    if (tracks_[i]->foundTrack(simtrk, phimin_)) {
+      Tracklet* tracklet = tracks_[i];
+      int charge = simtrk.trackid() / abs(simtrk.trackid());
+      if (abs(simtrk.trackid()) < 100)
+        charge = -charge;
+      double simphi = simtrk.phi();
+      if (simphi < 0.0)
+        simphi += 2 * M_PI;
+      int irinv = tracklet->irinvfit().value();
+      if (irinv == 0)
+        irinv = 1;
+      int layerordisk = -1;
+      if (tracklet->isBarrel()) {
+        layerordisk = tracklet->layer();
+      } else {
+        layerordisk = tracklet->disk();
+      }
+      outres << layerordisk << " " << tracklet->nMatches() << " " << simtrk.pt() * charge << " " << simphi << " "
```

Comment:
is this run in "production" mode, or only for debug / development ?

Reply:
either way, I would suggesto replace `(0.3 * 3.8 / 100.0)` with a named constant

Reply:
no all file writing is just for non-production running (e.g. writing out memory content or lockup tables for the firmware). this particular part of the code has regardless now been removed to clean up and simplify, and using named constants for c/bfields throughout.